cli/testflinger_cli/client.py

```diff
@@ -112,19 +143,31 @@ def post_job_state(self, job_id, state):
         data = {"job_state": state}
         self.put(endpoint, data)
 
-    def submit_job(self, job_data):
+    def submit_job(self, data: dict) -> str:
         """Submit a test job to the testflinger server
 
         :param job_data:
-            String containing json or yaml data for the job to submit
+            Dictionary containing json or yaml data for the job to submit
```

Comment:
The string contains YAML (or JSON, as YAML is compatible with it), but the dictionary is just a data structure, and doesn't "contain" YAML or JSON itself (it will be serialized as JSON during the request).

This might be seen as nitpick-y, but it would be more correct as:

```suggestion
            Dictionary containing the job to submit
```